File: 2021/2-2. Dive/C++/src/main.cpp
#include <fstream>
#include <iostream>
#include <string>

int main()
{
    std::ifstream infile("input.txt");

    if (!infile.is_open())
    {
        std::cerr << "ERROR! Could not open input.txt!\n";
        return -1;
    }

    int position = 0;
    int depth = 0;
    int aim = 0;

    std::string type;
    int ammount;

    while (infile >> type >> ammount)
    {
        if (type == "forward")
        {
            position += ammount;
            depth += ammount * aim;
        }else if (type == "up"){
            aim -= ammount;
        }else{
            aim += ammount;
        }
    }

    std::cout << "Answer: " << position * depth << '\n';

    return 0;
}

File: 2021/2-2. Dive/Python/main.py
infile = open("input.txt")

pos = 0
depth = 0
aim = 0

for line in infile:
    command, ammount = line.split()

    ammount = int(ammount)

    if command == "forward":
        pos += ammount
        depth += ammount * aim
    elif command == "up":
        aim -= ammount
    else:
        aim += ammount

print("Answer:", pos * depth)

infile.close()